version: '3.9'

services:

  cellenium:
    build:
      context: .
      dockerfile: Dockerfile
    env_file:
      - '.env'
    volumes:
      - ${LOCAL_CREDENTIALS}:${CONTAINER_CREDENTIALS}
    ports:
      - ${CORE_PORT}
    healthcheck:
      test: [ "CMD-SHELL", "python --version || exit 1" ]
      interval: 30s
      timeout: 10s
      retries: 2
    command: >
      bash -c "pytest -n ${PARALLEL} && allure generate allure-results --clean -o allure-results && allure serve"

    deploy:
      resources:
        limits:
          cpus: '2'

  allure-server:
    image: frankescobar/allure-docker-service:latest
    volumes:
      - ./allure-results:/cellenium/allure-results
    environment:
      CHECK_RESULTS_EVERY_SECONDS: 10
      KEEP_HISTORY: 1
      ANALYTICS: 1
      SECURITY: 0
    ports:
      - ${ALLURE_PORT}
    healthcheck:
      test: ["CMD", "allure --version || exit 1"]
      interval: 10s
      retries: 6
    restart: on-failure
    depends_on:
      - cellenium
#      cellenium:
#        condition: service_completed_successfully
